### DISCUSSION TITLE
RangeSelector1D object has no attribute 'copy' in create_scool

### DISCUSSION THREAD

**Discussion by bskubi** [open2c/cooler#395] [Q&A] [answered]
```
      cooler.create_scool(scool_file, bins, pixels)
    File "[...]/python3.12/site-packages/cooler/create/_create.py", line 1174, in create_scool
      bins = bins.copy()
             ^^^^^^^^^
  AttributeError: 'RangeSelector1D' object has no attribute 'copy'
```

I'm tinkering around with the create_scool function in the cooler library, but it gives me this error.

```
bins = {}
    pixels = {}
    for id, cool in zip(block_ids, cools):
        c = cooler.Cooler(f"{cool}::/")
        bins[id] = c.bins()
        pixels[id] = c.pixels()
    cooler.create_scool(scool_file, bins, pixels)
```

The `cool` files exist. I'm just validating a workflow with some small test data here, so I can't guarantee they're not corrupted, but I can use zoomify to make mcools of them successfully.

**Comment by nvictus** [comment, marked as answer]
The bin and pixel tables need to be materialized into dataframes (calling `c.bins()` and `c.pixels()` return "selectors"). 

```python
bins[id] = c.bins()[:]
pixels[id] = c.pixels()[:]
```

**Comment by bskubi** [comment]
Problem solved, thank you!
